soc/arm/nxp_imx/rt/Kconfig.soc

```diff
@@ -246,4 +246,30 @@ config DEVICE_CONFIGURATION_DATA
 
 endif # NXP_IMX_RT_BOOT_HEADER
 
+choice CODE_LOCATION
+	prompt "Code location selection"
+	default CODE_FLEXSPI
+
+config CODE_ITCM
+	bool "Link code into internal instruction tightly coupled memory (ITCM)"
+
+config CODE_FLEXSPI
+	bool "Link code into external FlexSPI-controlled memory"
+	select NXP_IMX_RT_BOOT_HEADER
+
+endchoice
+
+choice DATA_LOCATION
+	prompt "Data location selection"
+	default DATA_SEMC
```

Comment:
Is it fine defaulting to SDRAM at the SoC level? All the EVK boards have SDRAM, but not all boards necessary have one.

Reply:
My original thought was the most common case should be the default, but you make a good point about non-EVK boards. Will rework the defaults such that TCM is default at the SoC level and EVK boards override them.